src/connection/manager.ts

```diff
@@ -117,10 +117,13 @@ export class SQLJobManager {
     const selected = this.jobs[this.selectedJob]
     if (ServerComponent.isInstalled() && selected) {
       return selected.job.query<T>(query, opts);
-    } else if(!ServerComponent.isInstalled()) {
+    } else if (!ServerComponent.isInstalled()) {
       let updateResult = await ServerComponent.checkForUpdate();
-      if(UpdateStatus.JUST_UPDATED === updateResult) {
-        setTimeout(() => {JobManagerView.setVisible(true)});
+      if (UpdateStatus.JUST_UPDATED === updateResult) {
+        JobManagerView.setVisible(true);
+        if (`new`=== Configuration.get<string>(`alwaysStartSQLJob`)) {
```

Comment:
For `alwaysStartSQLJob` to be set to `new` rather than `ask`, the user has to have installed the server component at some point prior.  Upon installation they would have been prompted to start the server job, and would have had to answer Always.

Given that, consider my scenario.

At startup, I got the prompt to install the server component, but I cancelled it, :smirk: then proceeded to try to run some SQL, and got the prompt to install again.  This time the install was performed, but the server job wasn't started automatically because the setting is still set to `ask`.  I still got the `Active SQL job is required. Please spin one up in the 'SQL Job Manager' view and try again.` message.

Could it have presented the prompt rather than the error message?
![image](https://github.com/halcyon-tech/vscode-db2i/assets/134640655/b12a8487-aac2-42a7-863f-9862844bbce2)

Reply:
resolved at https://github.com/halcyon-tech/vscode-db2i/pull/74/commits/090952b7e35ae5fd0b607851ad95b87984251ba5